### PR TITLE
Feature: setup default error management

### PR DESCRIPTION
**Pull request by Zorin95670**
Make error managment for the api.

### CODE REVIEW

File: src/main/java/com/ditrit/letomodelizerapi/model/error/ErrorDTO.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+import lombok.Data;
+
+import java.io.Serial;
+import java.io.Serializable;
+
+/**
+ * Model of error.
+ *
+ * @author Vincent Moittié
```
Comment (justinehell) [inline src/main/java/com/ditrit/letomodelizerapi/model/error/ErrorDTO.java:11]: remove

File: src/main/java/com/ditrit/letomodelizerapi/model/error/ErrorType.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+     */
+    WRONG_FILTER_OPERATOR(203, "Field contains a wrong operator.", HttpStatus.BAD_REQUEST),
+    /**
+     * Error to call when entity is not found in database.
+     */
+    ENTITY_NOT_FOUND(204, "Entity not found.", HttpStatus.NOT_FOUND),
+    /**
+     * Error to call when there are problems with field.
+     */
+    WRONG_FIELD(205, "Somethings wrong with this field.", HttpStatus.BAD_REQUEST),
```
Comment (justinehell) [inline src/main/java/com/ditrit/letomodelizerapi/model/error/ErrorType.java:38]: Something is wrong with this field.

File: src/main/java/com/ditrit/letomodelizerapi/model/error/package-info.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+/**
+ * Package that contains all models that related to Error like Exception or DTO/Record.
```
Comment (aboubakar14) [inline src/main/java/com/ditrit/letomodelizerapi/model/error/package-info.java:2]: Package containing all models related to Errors like Exception or DTO/Record errors.

File: src/main/java/com/ditrit/letomodelizerapi/model/error/ErrorDTO.java (hunk 1 above)
Comment (aboubakar14) [inline src/main/java/com/ditrit/letomodelizerapi/model/error/ErrorDTO.java]: To remove

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -42,6 +42,25 @@ Once database is setup you can run :
...
 
+## Run application tests
+
+To run all the application tests (unit and integration), use this command:
+
+```shell
+./gradlew test -i
+```
+
+To run only  the application unit tests, use this command:
```
Comment (justinehell) [inline CONTRIBUTING.md:53]:
```suggestion
To run only the application unit tests, use this command:
```

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/main/java/com/ditrit/letomodelizerapi/model/error/ApiException.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    private final ErrorDTO error;
+    /**
+     * Log exception.
+     */
+    private final boolean logIt;
+
+    /**
+     * Parameters error exception.
+     *
+     * @param error Error's type.
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/main/java/com/ditrit/letomodelizerapi/model/error/ApiException.java:35]: done

File: src/main/java/com/ditrit/letomodelizerapi/model/error/ErrorDTO.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    /**
+     * Constructor that init all member.
+     *
+     * @param code Error code.
+     * @param message Error message.
+     * @param field   Error field.
+     * @param value   Error value.
+     * @param cause   Error exception Cause.
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/main/java/com/ditrit/letomodelizerapi/model/error/ErrorDTO.java:56]: done

File: src/main/java/com/ditrit/letomodelizerapi/model/error/ErrorType.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+     * Exception status.
+     */
+    private final HttpStatus status;
+
+    /**
+     * Error type with message.
+     *
+     * @param code Error code.
+     * @param message Error message.
+     * @param status  Error status.
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/main/java/com/ditrit/letomodelizerapi/model/error/ErrorType.java:73]: done

File: src/main/java/com/ditrit/letomodelizerapi/model/error/ErrorDTO.java (hunk 1 above)
Comment (Zorin95670) [reply src/main/java/com/ditrit/letomodelizerapi/model/error/ErrorDTO.java:11]: done

File: CONTRIBUTING.md (hunk 4 above)
Comment (Zorin95670) [reply CONTRIBUTING.md:53]: done

File: src/main/java/com/ditrit/letomodelizerapi/controller/handler/ConstraintViolationExceptionHandler.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+            return Response.status(status).entity(mapper.writeValueAsString(error))
+                    .type(MediaType.APPLICATION_JSON)
+                    .build();
+        } catch (JsonProcessingException e) {
+            throw new ClientErrorException(Response.Status.INTERNAL_SERVER_ERROR, e);
+        }
+    }
+
+    /**
+     * Get last path.
```
Comment (aboubakar14) [inline src/main/java/com/ditrit/letomodelizerapi/controller/handler/ConstraintViolationExceptionHandler.java:64]: Better doc please

File: src/main/java/com/ditrit/letomodelizerapi/model/error/package-info.java (hunk 3 above)
Comment (Zorin95670) [reply src/main/java/com/ditrit/letomodelizerapi/model/error/package-info.java:2]: done

File: src/main/java/com/ditrit/letomodelizerapi/model/error/ErrorDTO.java (hunk 1 above)
Comment (Zorin95670) [reply src/main/java/com/ditrit/letomodelizerapi/model/error/ErrorDTO.java]: done

**Review by aboubakar14 — APPROVED** [review]

**Pull request merged**